kuka_rsi_hw_interface/krl/README.md

```diff
@@ -1,4 +1,93 @@
+
+1. Log in as **Administrator** on the teach pad and navigate to **Network configuration** (Main menu -> Start-up -> Network configuration -> Activate advanced configuration).
+2. There should already be an interface checked out as the **Windows interface**.
+   * **IP**: e.g. 192.168.250.20
+   * **Subnet mask**: 255.255.255.0
+   * **Default gateway**: 192.168.250.20
+   * **Windows interface checkbox** should be checked.
+3. Minimize the SmartHMI (Main menu -> Start-up -> Service -> Minimize HMI).
+4. Run the application called **RSI-Network** from the Windows desktop environment on the teach pad.
+5. Check that the **Network - Kuka User Interface** show the Windows interface with the specified IP address.
+6. Add a new IP address on another subnet (e.g. 192.168.1.20) for the **RSI interface**.
+   * This should be possible within **RSI-Network**, if not, then do it within the SmartHMI at **Network configuration**.
+7. Reboot the controller (Main menu -> Shutdown -> Check **Force cold start** and **Reload files** -> Reboot control PC).
+8. After reboot, minimize the SmartHMI (Main menu -> Start-up -> Service -> Minimize HMI).
+9. Run **cmd.exe** and ping the PC you want to communicate with on the same subnet (e.g. 192.168.250.xx).
+
+If your **PC** has an IP address on the same subnet as the **Windows interface** on the controller, the controller should receive answers from the PC:
+* If this is the case, add another IP address to the current PC connection (e.g. 192.168.1.xx) on the same subnet as the **RSI** interface.
+
+## 2. KRL Files
+
+The files included in this folder specifies the data transferred via RSI. Some of the files needs to be modified to work for your specific configuration.
+
+##### ros_rsi_ethernet.xml
+1. Edit the `IP_NUMBER` tag so that it corresponds to the IP address (192.168.1.xx) previously added for your PC.
+2. Keep the `PORT` tag as it is (49152) or change it if you want.
+
+##### ros_rsi.rsi.xml
+This file may be edited with application specific joint limits.
+* Edit the parameters within the RSIObject `AXISCORR` to specify joint limits such as **LowerLimA1, UpperLimA1** etc.
+* Edit the parameters within `AXISCORRMON` to specify the amount of degrees in which the robot should have reached its goal. The values of **MaxA1, MaxA2** etc. may be large to allow free movement within the specified joint limits in `AXISCORR`.
+
+Notice the RSIObject of type `ETHERNET`. Within this object is a parameter called **Timeout**. This parameter is set to **100** by default. The RSI interface operates at 250 Hz and expects new data every 4ms. If the connected **PC with ROS** does not fulfill this, a message is missed and a counter is incremented. When this counter hits **100**, the RSI connection will shut down. If you have problems with the connection to RSI shutting down now and then while moving the robot it is suggested to:

```

Comment:
> The RSI interface operates at 250 Hz and expects new data every 4ms.

might be good to mention here that the `hw_iface` doesn't have a period configured, and is completely driven by the controller's output.